b2b_ecommerce/models.py

```diff
@@ -38,15 +38,13 @@ def get_unexpired_coupon(self, *, coupon_code, product_id):
                 The coupon instance. If no coupon is found a B2BCoupon.DoesNotExist error is raised
         """
         return (
-            self.filter(coupon_code=coupon_code, product_id=product_id, enabled=True)
+            self.filter(
+                Q(coupon_code=coupon_code),
+                Q(enabled=True),
+                Q(product_id=None) | Q(product_id=product_id),
+            )
             .filter(Q(activation_date__isnull=True) | Q(activation_date__lt=Now()))
             .filter(Q(expiration_date__isnull=True) | Q(expiration_date__gt=Now()))
-            .exclude(
```

Comment:
I think we need a flag on `B2BCoupon` so we can distinguish between coupons which allow multiple use and those which don't. We don't want to change the behavior of existing coupons, only newly created ones.

Reply:
@briangrossman do we need to need to keep the old behavior of existing coupons too?

Reply:
@umarmughal824 
Good question.
It's most important to get the new functionality working. 
Only keep the old behavior if it's very easy.

Reply:
Sure, will fix it tomorrow.